src/net/posix-stack.cc

```diff
@@ -998,8 +998,8 @@ std::vector<network_interface> posix_network_stack::network_interfaces() {
             bool done = false;
 
             while (!done) {
-                msghdr rtnl_reply = { 0, };
-                iovec io_reply = { 0, };
+                msghdr rtnl_reply = {};
+                iovec io_reply = {};
```

Comment:
Both rtnl_reply and io_reply are initialized with some values few lines below, I think it's worth just initialize both with those values here instead of zeroifying

Reply:
i concur with you on dropping the zero-initialization of  `io_reply`. but i am not quite sure about `rtnl_reply`. i agree that the syscall does not read `[msg_control](https://docs.huihoo.com/doxygen/linux/kernel/3.7/structmsghdr.html#ae0bcd626259df3c2f039cee9353150c6)` or [msg_controllen](https://docs.huihoo.com/doxygen/linux/kernel/3.7/structmsghdr.html#a9d26eb73cb060c84f34a9e1aa58e631e) fields in this case, but i don't want to change the existing behavior in this refactor. -- especially, i still don't have an environment for testing the DPDK application at this moment. 

regarding `io_reply`, i created another commit to use designated initialization. so that the first commit can focus on the preserving the existing code structure  in addition to its behavior.

Reply:
The `rtnl_reply` is initialized below:

```
                 rtnl_reply.msg_iov = &io_reply;
                 rtnl_reply.msg_iovlen = 1;
                 rtnl_reply.msg_name = &kernel;
                ...
```

you can patch it to be initialized with designated initializers instantly

Reply:
yes, i can. but we have to initialize the members *in the order of* their declarations. if what you suggest is to initialize `msg_control` and `msg_controllen` as well. well, that's an option also. i don't see the merit of doing so in comparison with the zero initialization. but if you still think we should go this way, i will implement in the next revision.